### PR TITLE
Introduce message.RoleMap.

### PR DESCRIPTION
**Pull request by jmalloc**
This is a data structure that maps a message type to its role.  This is needed often throughout the config system.

It conforms to a common interface with the existing `TypeSet`, allowing the two to be used interchangeably by code that only needs to test if a given message type is present.

### CODE REVIEW

File: message/rolemap.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package message
+
+import (
+	"github.com/dogmatiq/dogma"
+)
+
+// RoleMap is a map of message type to role.
+// It implements the TypeContainer interfacer.
+type RoleMap map[Type]Role
```
Comment (danilvpetrov) [inline message/rolemap.go]: quick question: should it be concurrently safe? The reason I am asking is that some parts of `enginekit` have concurrent logic.

Comment (jmalloc) [reply message/rolemap.go]: I'll leave it up to the code *building* the map to worry about concurrency.  The typical use case would be to build it once at config stage and then use it read-only after that.

File: message/rolemap.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package message
+
+import (
+	"github.com/dogmatiq/dogma"
+)
+
+// RoleMap is a map of message type to role.
+// It implements the TypeContainer interfacer.
```
Comment (koden-km) [inline message/rolemap.go]: Typo? interface**r**
> interfacer

File: message/container.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package message
+
+import (
+	"github.com/dogmatiq/dogma"
+)
+
+// TypeContainer is an interface for contains of message types.
+type TypeContainer interface {
+	// Has returns true if the set contains t.
```
Comment (koden-km) [inline message/container.go]: These methods mention *set* but it is not specific to a set, not even by interface name?

**Pull request merged**